### ISSUE TITLE
Using returned strings from wake within shell scripts

### ISSUE THREAD

**Issue by mmjconolly** [sifive/wake#1037]
Sometimes we want to extract a particular value defined within wake code for usage outside of wake. For example, the version number of a tool that is used.

If we run commands like these from the command line:
```
$ wake -x 'myFuncReturnsAString foo'
"i am a string"
```
We get quotes around the response.
It would be nice if we had a `--raw-string` or similar argument to not produce the quotes and no other output except the `String` returned by the function for usage within shell scripts.

If we try to use this within a shell currently we see:
```
$ echo $(wake -x 'myFuncReturnsAString foo')
wake: targeting utilization for 1.0 threads and 1GiB of memory. "i am a string"
```
which shouldn't print anything other than `i am a string`, such as this potential future invocation:
```
$ echo $(wake --raw-string -x 'myFuncReturnsAString foo')
i am a string
```

**Comment by V-FEXrt** [comment]
@mmjconolly Is the idea that a string will always be returned? Is it okay to error on a result for example?

**Comment by mmjconolly** [comment]
I think we have 3 cases perhaps
* myfunc -> String
* myfunc -> Result String Error
* myfunc -> Option String

I'm thinking that if we get returned an `Error` or a `None` (or some other type), that wake returns a non-zero shell exit code and prints an error to stderr

**Comment by V-FEXrt** [comment]
Makes sense, I think that should be doable. I'm mostly worried about the `wake: targeting utilization for 1.0 threads and 1GiB of memory. ` part because I think that is emitted due to a separate flag but we'll see. The flags in main are all already quite intertwined

**Comment by mmjconolly** [comment]
ah yeah the `wake: targeting utilization` shouldn't be printed

**Comment by V-FEXrt** [comment]
I think we resolved this via streams. I'm going to go ahead and close but feel free to re-open if you disagree

**Comment by mmjconolly** [comment]
Whats an example of how streams could be used for this?

**Comment by V-FEXrt** [comment]
You redirect the string you care about to a stream, them redirect that stream to stdout, and use `-q` to turn off all the other output.

something like:
```
> echo $(./bin/wake --stdout report -q -x 'printlnLevel logReport "asdf"')
asdf
```

**Comment by mmjconolly** [comment]
It looks like `--log-header ''` `--no-tty`  `3>&1 1>/dev/null` need to be added to these to make them reliable.

It may be useful to have a mode where it presents precisely/exactly/only the requseted stream with all other printing disabled so that the user doesn't have to know these details

**Comment by mmjconolly** [comment]
After some discussion, I feel it's important to provide an explict option, like `--raw-string` so that future changes to logging wont impact these usecases.
You will never prevent people from writing shell scripts to extract information from wake builds, so we need to make it low-impact to maintain
